api/helpers/gamestate/moves/draw/execute.js

```diff
@@ -30,7 +30,7 @@ module.exports = {
 
     const player = playedBy ? result.p1 : result.p0;
 
-    player.hand.push(currentState.deck.shift());
+    player.hand.push(result.deck.shift());
```

Comment:
Whoopsie daisy. Good catch, thank you. This highlights a shortcoming in the tests since the test for drawing a card doesn't check the number of cards in the deck. That might be worth addressing later on. For now, nice fix